src/engraving/playback/playbackcontext.cpp

```diff
@@ -293,6 +303,61 @@ void PlaybackContext::handleAnnotations(const ID partId, const Segment* segment,
     }
 }
 
+void PlaybackContext::handleMeasureRepeats(const std::vector<const MeasureRepeat*>& measureRepeats, const int tickPositionOffset)
+{
+    for (const MeasureRepeat* mr : measureRepeats) {
+        const Measure* currMeasure = mr->firstMeasureOfGroup();
+        if (!currMeasure) {
+            continue;
+        }
+
+        const Measure* referringMeasure = mr->referringMeasure(currMeasure);
+        if (!referringMeasure) {
+            continue;
+        }
+
+        int newDynamicsOffsetTick = mr->tick().ticks() + tickPositionOffset - referringMeasure->tick().ticks();
+
+        for (int num = 0; num < mr->numMeasures(); ++num) {
+            int startTick = referringMeasure->tick().ticks();
+            int endTick = referringMeasure->endTick().ticks();
+
+            copyDynamicsInRange(startTick, endTick, newDynamicsOffsetTick);
+
+            currMeasure = currMeasure->nextMeasure();
+            if (!currMeasure) {
+                break;
+            }
+
+            referringMeasure = mr->referringMeasure(currMeasure);
+            if (!referringMeasure) {
+                break;
+            }
+        }
+    }
+}
+
+void PlaybackContext::copyDynamicsInRange(const int rangeStartTick, const int rangeEndTick, const int newDynamicsOffsetTick)
+{
+    auto startIt = m_dynamicsMap.find(rangeStartTick);
+    if (startIt == m_dynamicsMap.end()) {
+        return;
+    }
```

Comment:
This combination of `find` and `return` seems to break this case: 
<img width="281" alt="Scherm­afbeelding 2023-12-20 om 21 01 14" src="https://github.com/musescore/MuseScore/assets/48658420/a0cbf256-878b-4439-85e7-7a08bb9b9588">
(when it's not the first measure)

It looks like replacing `find` with `lower_bound` gives the correct result.